# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1289229577 -28800
#      Mon Nov 08 23:19:37 2010 +0800
# Node ID 83d268af65cbc84754e33f52e32eaf828affe5be
# Parent  abee8cc4820ff95a93529aa17aae87df1f3dd70b
__go_draw_axes__.m: Check for z/y/zdata before converting negative values to NaNs.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2010-11-08  Ben Abbott <bpabbott@mac.com>
 
+	* plot/__go_draw_axes__.m: Check for z/y/zdata before converting
+	negative values to NaNs. Bug fix for changeset 11201.
 	* plot/daspect.m: Fix demo using both daspect and pbaspect.
 
 2010-11-08  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Remove foreground color earlier.
 	* plot/__go_draw_figure__.m: Don't set the foreground color here.
 
 2010-11-08  Rik  <octave@nomad.inbox5.com>
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -401,23 +401,23 @@ function __go_draw_axes__ (h, plot_strea
         end_unwind_protect
       endif
       kids = kids(1:(end-1));
 
       if (strcmpi (obj.visible, "off"))
         continue;
       endif
 
-      if (xlogscale)
+      if (xlogscale && isfield (obj, "xdata"))
         obj.xdata(obj.xdata<=0) = NaN;
       endif
-      if (ylogscale)
+      if (ylogscale && isfield (obj, "ydata"))
         obj.ydata(obj.ydata<=0) = NaN;
       endif
-      if (zlogscale)
+      if (zlogscale && isfield (obj, "zdata"))
         obj.zdata(obj.zdata<=0) = NaN;
       endif
 
       ## Check for facecolor interpolation for surfaces.
       doing_interp_color = ...
          isfield (obj, "facecolor") && strncmp (obj.facecolor, "interp", 6);
 
       switch (obj.type)
